Breeder/BR.cpp

```diff
@@ -44,17 +44,35 @@
 /******************************************************************************
 * Globals                                                                     *
 ******************************************************************************/
-static COMMAND_T* g_nextActionApplyer = NULL;
-static int        g_nextActionToApply = 0;
-static int        g_nextActionLoCmd   = 0;
-static int        g_nextActionHiCmd   = 0;
+static bool       g_deferRefreshToolbar = false;
+static COMMAND_T* g_nextActionApplyer   = NULL;
+static int        g_nextActionToApply   = 0;
+static int        g_nextActionLoCmd     = 0;
+static int        g_nextActionHiCmd     = 0;
 static set<int>   g_nextActionApplyers;
 
 /******************************************************************************
 * Command hook                                                                *
 ******************************************************************************/
+static void DeferRefreshToolbar()
```

Comment:
Is there a reason for waiting two timer cycles before refreshing the toolbars? I'd expect a function like this to remove the timer at the next cycle without an extra flag to toggle and check.

Reply:
Yes, it is needed because it needs to refresh AFTER MIDI editor opens...

Reply:
Isn't the MIDI editor being opened synchronously when the action 40153 is run? (so always before the next defer cycle)

I added a breakpoint in `DeferRefreshToolbar` and it's always hit after the ME has opened. Also `MIDIEditor_GetActive` returns non-null the first time `DeferRefreshToolbar` is called.

Reply:
When I tested it, it didn't work until I added this.